okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -457,13 +495,27 @@ class RealWebSocket(
       if (pong != null) {
         writer!!.writePong(pong)
       } else if (messageOrClose is Message) {
-        val data = (messageOrClose as Message).data
+        val uncompressedData = (messageOrClose as Message).data
+
+        var data = uncompressedData
+        var compressMessage = false
+        messageDeflater?.let { deflater ->
+          val deflated = deflater.deflate(uncompressedData).readByteString()
+          // If context takeover is NOT used, it's guaranteed that sending uncompressed message
+          // which is smaller than compressed version will be more efficient.
+          // We cannot make such assumptions if context takeover is used.
```

Comment:
Good comment.

Reply:
For simplicity I’d like to pay the ~14-18 bytes overhead and remove this check. I expect that everyone is going to use context takeover always, and so it’s complexity in our code to save 18 bytes that will probably never exist.

Reply:
(According to RFC 1979, the deflater won’t add more than 14-18 bytes of overhead on data that doesn’t compress well)